pkg/gatewayserver/gatewayserver_test.go

```diff
@@ -721,8 +859,15 @@ func TestGatewayServer(t *testing.T) {
 						if !a.So(err, should.BeNil) {
 							t.FailNow()
 						}
-						a.So(stats.UplinkCount, should.Equal, uplinkCount)
-						if tc.Up.GatewayStatus != nil {
+
+						if tc.StatusMsgSupported {
+							a.So(stats.UplinkCount, should.Equal, uplinkCount)
+						} else {
+							// For protocols that doesn't support GatewayStatus messages, there will be one less uplink.
+							a.So(stats.UplinkCount, should.BeBetweenOrEqual, uplinkCount-2, uplinkCount)
```

Comment:
Make this deterministic. Can we solve this with an `ExpectedUplinkCount` in the inner test table?